gammapy/maps/hpx.py

```diff
@@ -1732,6 +1732,31 @@ def __repr__(self):
         str_ += "\tcenter     : {lon:.1f} deg, {lat:.1f} deg\n".format(lon=lon, lat=lat)
         return str_
 
+    def __eq__(self, other):
+        """Test equality between two `~gammapy.maps.HpxGeom`"""
+        if not isinstance(other, self.__class__):
+            raise TypeError('Cannot compare HpxGeom with {}'.format(other.__class__))
+
+        if self._sparse or other._sparse:
+            raise ValueError("sparse geometries not supported")
+        if self.is_allsky and other.is_allsky is False:
+            raise ValueError("Non allsky HpxGeom not supported")
+
+        # check overall shape and axes compatibility
+        if self.data_shape != other.data_shape:
+            return False
+
+        result = True
```

Comment:
This is a little simpler, no?
```
for ax1, ax2 in zip(self.axes, other.axes):
    if ax1 != ax2:
        return False
```
Alternatively if you want to create a local status variable, I think this would be equivalent:
```
axes_equal = all(ax1 == ax2 for ax1, ax2 in zip(self.axes, other.axes))
```

Reply:
OK. Changed to first solution